lib/BackgroundJob/RetentionJob.php

```diff
@@ -104,10 +104,12 @@ public function run($argument) {
 		} catch (\InvalidArgumentException $e) {
 			// tag is invalid remove backgroundjob and exit
 			$this->jobList->remove($this, $argument);
+			$this->logger->warning("Background job was removed, because tag $tag is invalid");
```

Comment:
sorry for not noticing earlier but a little improvement would be to log the exception as well, because the trace typically helps a lot troubleshoot a problem. Could you change from `->warning(...)` to `->logException($e, ['message' => 'Background job ...', 'level' => ILogger::DEBUG])`?

same for the other two logs below :)